spec/modularity/as_trait_spec.rb

```diff
@@ -201,6 +201,35 @@ def vanilla_method
 
       end
 
+      it 'allows keyword args in all supported ruby versions' do
```

Comment:
```ruby
it 'passes keyword args to the block given to as_trait`
```

Reply:
> However, when we introduce non-trivial version switches like this, our tests should also run against multiple Ruby versions with GitHub Actions.

We already use GitHub Actions to test multiple Ruby versions (see "checks" tab of this PR or [here](https://github.com/makandra/modularity/blob/master/.github/workflows/test.yml))
It's not tested against Ruby 3.1 though, so i will add that.